lib/Twig/Extension/Core.php

```diff
@@ -901,9 +923,25 @@ function twig_in_filter($value, $compare)
     } elseif (is_string($compare) && (is_string($value) || is_int($value) || is_float($value))) {
         return '' === $value || false !== strpos($compare, (string) $value);
     } elseif ($compare instanceof Traversable) {
-        return in_array($value, iterator_to_array($compare, false), is_object($value) || is_resource($value));
+        if (is_object($value) || is_resource($value)) {
+            foreach ($compare as $key => $item) {
+                if ($item === $value) {
+                    return true;
+                }
+            }
+        } else {
+            foreach ($compare as $key => $item) {
+                if ($item == $value) {
+                    return true;
+                }
+            }
+        }
+
+        return false;
     }
 
+    @trigger_error('Using "in" must be done on an array, string or Traversable, deprecated since version 1.29 and will be removed in Twig 2.0', E_USER_DEPRECATED);
```

Comment:
the message is not easy to understand IMO. It is not clear about what is deprecated